In [1]:
from math import factorial

<h4> Задача 1</h4>

Группа студентов изучает 10 различных дисциплин. Сколькими способами можно составить расписание на понедельник, если в этот день должно быть 4 разных занятия?

Т.к. нам нужны только 4 предмета из 10 и нам не важно каким будет расположение друг относительно друга, значит это способ сочетаний

In [2]:
def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

In [3]:
combinations(10, 4)

210

**Задача 2**

Из колоды в 52 карты вынимают случайным образом 4 карты. Найти число исходов, соответствующих тому, что был вытянут хотя бы один туз.

Чтобы решить эту задачу, нужно сначала рассмотреть ситуацию, когда выбирается 1 туз из четырех.

Число таких сочетаний будет равно:

In [4]:
combinations(4, 1)

4

Остальные три карты выбираются из 48 карт  — тузы не рассматриваются, так как они уже выбраны.Таких  сочетаний получилось:

In [5]:
combinations(48, 3)

17296

И тогда будет такое число сочетаний, когда из колоды в 52 карты выбраны четыре, одна из которых — туз:

In [6]:
combinations(4, 1)*combinations(48, 3)

69184

**Задача 3**

Семь человек рассаживаются наудачу на скамейке. Какова вероятность того, что два определённых человека будут сидеть рядом?

 Считаю, что задача является совместной, значие из суммы вероятностей отдельных событий вычитается вероятность их совместного появления.

$$P(А+B) = \frac{1}{7} + \frac{1}{7} - (\frac{1}{7}*\frac{1}{7}) = \frac{2}{7} - \frac{1}{14} = \frac{3}{14} \approx 0.214$$


**Задача 4**

Из 60 вопросов, входящих в экзаменационные билеты, студент знает 50. Какова вероятность того, что среди трёх наугад выбранных вопросов студент знает: а) все? б) два?

**a)** Разобьем 60 билетов на 50 - которые сутдент знает, и 10 - оставшиеся, которые не знает. Нужно узнать вероятность, что среди выбранных он знает все, т.е. из 50 билетов выбрать 3. Вероятность будем находить по фрмуле: $$P(A) = \frac{m}{n}$$



для этого, найдем общее число событий по фрмуле

$$C_n^m = \frac{n!}{m!(n-m)!}$$, получаем $$n = C_{60}^3 = \frac{60!}{3!(60-3)!} = 34220$$ или

In [7]:
combinations(60, 3)

34220

(m\)  - число элементарных исходов, благоприятствующих событию \(A\), будем искать по формуле сочетаний $$m = C_{50}^3 = \frac{50!}{3!47!} = 19600$$

In [8]:
combinations(50, 3)

19600

подставляем в формулу вероятности $$P(A) = \frac{m}{n} = \frac{19600}{34220} = 0.573$$


***Ответ (а):*** вероятность того, что среди наугад выбранных трёх вопросов студент знает все три равна $$P(A) = 0.573$$ 

б) Таким же образом найдем верояность для 2 билетов из 50, которые студент знает. **и** для 1 из 10, которые не знает.Таким образом получится:

(m\) - число элементарных исходов, благоприятствующих событию для 2 карт из тех что знает равно:

In [9]:
combinations(50, 2)

1225

число исходов, для 1 карты из тех что не знает, равно:

In [10]:
combinations(10, 1)

10

находим произведение исходов:

In [11]:
m = combinations(50, 2) * combinations(10, 1)
print(m)

12250


подставляем в формулу вероятности $$P(A) = \frac{m}{n} = \frac{12250}{34220} = 0.35798$$

***Ответ (б):*** вероятность того, что среди наугад выбранных трёх вопросов студент знает 2 равна $$P(A) = 0.358$$ 

**Задача 5**

Бросается игральная кость. Пусть событие A - появление чётного числа, событие B - появление числа больше трёх. Являются ли эти события независимыми?

*считаю, что события являются зависимыми, т.к. четное число и число больше 3 может произойти одновременно на одном кубике, например, может выпасть число 4 или 6*

чтобы проверить найдем Р(А) и Р(А|B), если значения будут равны, значит события независимые.

$$P(A) = \frac{1}{6} + \frac{1}{6} + \frac{1}{6} = \frac{1}{2} = 0.5$$

числа больше трех(4,5,6) для каждого числа вероятность будет 1/6. Предположим, что собыите А произошло и выпало четное число


$$P(В|A) = \frac{1}{6} + \frac{1}{6} = \frac{1}{3} \approx 0.33$$

**Ответ:** события являются зависимыми

**Задача 6 (Дополнительно)**

Допустим, имеется некоторая очень редкая болезнь (поражает 0.1 % населения). Вы приходите к врачу, вам делают тест на эту болезнь, и тест оказывается положительным. Врач говорит вам, что этот тест верно выявляет 99 % больных этой болезнью и всего лишь в 1 % случаев даёт ложный положительный ответ.

Вопрос: какова вероятность, что вы действительно больны ей?

Подсказка: вновь используйте формулу Байеса с раскрытием знаменателя с помощью формулы полной вероятности.

Предположим, что событие А - пациент заболел. Для каждого человека возможность зарожения составляет P(A|B) = 1/1000 или 0,1%. P(B1) - вероятность что пациент болен 99/100 или 99%, а вероятность, что тест ложный Р(В2)=1/100.

$$P(B\:|\:A) = \frac{P(A\:|\:B) \cdot P(B)}{P(A)}$$

$$P(A) = P(B_1) \cdot P(A | B_1) + P(B_2) \cdot P(A | B_2) + P(B_3) \cdot P(A | B_3)$$

подставляем наши значения
$$P(A) = \frac{99}{100} \cdot \frac{1}{1000} + \frac{1}{100} \cdot \frac{1}{1000} = \frac{1}{1000} $$

вероятность, что пациент действительно болен равна:

$$P(B\:|\:A) = \frac{P(A\:|\:B) \cdot P(B)}{P(A)} = \frac{\frac{1}{1000} \cdot \frac{99}{100}}{\frac{1}{1000}} = \frac{99}{100} = 99 \% $$